### PR TITLE
NONE: change seed data date range

### PR DESCRIPTION
**Pull request by rickaw**
## What's changed
- change seed data date range
- Add at least one family member for each client
- Assign random gender for each family member

## Checklist
- [ ] The ticket is up-to-date - Please document any deviations from the original approach if there is any. - NA
- [ ] I have documented the testing steps for QA - NA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database...
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [ ] I have updated the typescript definitions for the database with `db:local:generate_types` - NA
  - [x] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - NO
  - [ ] I have checked that migration can happen successfully without resetting the database, doing the following. - NA
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

File: supabase/seed/dateData.mts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import seedrandom from "seedrandom";
 
-export const earliestDate = new Date(2024, 0, 1); // 2024/01/01
-export const latestDate = new Date(2025, 0, 1); // 2025/01/01
+export const earliestParcelEventDate = new Date(2024, 0, 1); // 2024/01/01
+export const latestParcelEventDate = new Date(2024, 4, 1); // 2024/05/01
```
Comment (AmeliaIsCoding) [inline supabase/seed/dateData.mts]: i would probs call this latestEventDate

File: supabase/seed/dateData.mts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import seedrandom from "seedrandom";
 
-export const earliestDate = new Date(2024, 0, 1); // 2024/01/01
-export const latestDate = new Date(2025, 0, 1); // 2025/01/01
+export const earliestParcelEventDate = new Date(2024, 0, 1); // 2024/01/01
```
Comment (AmeliaIsCoding) [inline supabase/seed/dateData.mts]: i would probs call this earliestParcelOrEventDate

File: supabase/seed/seed.mts [hunk 3]
```diff
@@ -59,19 +66,19 @@ async function generateSeed(): Promise<void> {
             extraInformation: (ctx) => copycat.sentence(ctx.seed, { maxWords: 20 }),
             flaggedForAttention: (ctx) => copycat.bool(ctx.seed),
             signpostingCallRequired: (ctx) => copycat.bool(ctx.seed),
+            families: (generateFamily) =>
+                generateFamily(
+                    { min: 1, max: 12 },
```
Comment (AmeliaIsCoding) [inline supabase/seed/seed.mts]: just pointing out I made the slightly arbitrary decision to generate family sizes between 1 and 12 inclusive but feel free to change the maximum number if you want

**Review by AmeliaIsCoding — COMMENTED** [review]
looks good, suggested better (imo) variable names but not sure if we care that much

File: supabase/seed/dateData.mts (hunk 1 above)
Comment (rickaw) [reply supabase/seed/dateData.mts]: changed

File: supabase/seed/dateData.mts (hunk 2 above)
Comment (rickaw) [reply supabase/seed/dateData.mts]: changed

File: supabase/seed/seed.mts (hunk 3 above)
Comment (rickaw) [reply supabase/seed/seed.mts]: changed to 8

**Review by AmeliaIsCoding — APPROVED** [review]
lgtm

**Pull request merged**